src/web/common.rkt

```diff
@@ -1,11 +1,16 @@
 #lang racket
-(require (only-in xml write-xexpr xexpr?) (only-in fpbench fpcore? core->c))
+(require (only-in xml write-xexpr xexpr?) 
+         (only-in fpbench fpcore? supported-by-lang? core->c core->tex expr->tex))
 (require "../common.rkt" "../syntax/read.rkt" "../programs.rkt" "../interface.rkt" "tex.rkt")
-(provide render-menu render-warnings render-large render-program program->fpcore render-reproduction)
+(provide render-menu render-warnings render-large render-program program->fpcore resugar-fpcore render-reproduction)
 
-(define (program->fpcore prog)
+(define (program->fpcore prog [transform identity])
```

Comment:
This API is pretty ugly; it's easier to just make `resugar-program` allow both programs and expressions, and then do the transform in the argument, instead of making it a callback.